server/routes/incidentDetails/incidentDetailsPage.ts

```diff
@@ -309,12 +312,18 @@ const extractIncidentDetails = (readDraftIncidentDetails: ExistingDraftIncidentD
 }
 
 const extractValuesFromPost = (req: Request): SubmittedFormData => {
+  let { discoveryDate } = req.body
+
+  if (req.body.discoveryRadioSelected === 'Yes') {
+    discoveryDate = req.body.incidentDate
+  }
+
   const values = {
     prisonerNumber: req.params.prisonerNumber,
     draftId: getDraftIdFromString(req.params.id),
     incidentDetails: {
       incidentDate: req.body.incidentDate,
-      discoveryDate: req.body.discoveryDate,
+      discoveryDate,
```

Comment:
As with one of the other comments - please try to not reassign things where it's not really necessary, especially data that's been gathered from user input.
This should be something like:
const { discoveryDate } = req.body
const discoveryDateTime = req.body.discoveryRadioSelected === 'Yes' ? req.body.incidentDate : discoveryDate
const values = {
...
discoveryDate: discoveryDateTime
...
}

Reply:
replaced with suggested